executor/executor_test.go

```diff
@@ -949,6 +949,8 @@ func (s *testSuite) TestUnion(c *C) {
 	tk.MustExec("create table t(a int)")
 	tk.MustExec("insert into t value(0),(0)")
 	tk.MustQuery("select 1 from (select a from t union all select a from t) tmp").Check(testkit.Rows("1", "1", "1", "1"))
+	tk.MustQuery("select 10 as a from dual union select a from t order by a desc limit 1 ").Check(testkit.Rows("10"))
```

Comment:
The bug this PR fixed is that, `order by a desc limit 1` belongs to union statement, but our old code thought 
it belongs to the second select statement.

Reply:
Yes.